File: CS426_final_project/app/src/main/java/com/example/cs426_final_project/api/FeedApi.java
package com.example.cs426_final_project.api;

import com.example.cs426_final_project.models.data.PostDataModel;
import com.example.cs426_final_project.models.posts.FeedResponse;
import com.example.cs426_final_project.models.posts.TimelineFeedResponse;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;

public interface FeedApi {
    @GET("feed/timeline")
    Call<String> getTimelineFeeds();

    @GET("feed/timeline")
    Call<List<TimelineFeedResponse> > getTimelineWidget();

}

File: CS426_final_project/app/src/main/java/com/example/cs426_final_project/api/FeedApi.kt
package com.example.cs426_final_project.api

import com.example.cs426_final_project.models.data.PostDataModel
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT

interface FeedApi {

    @GET("feed/timeline")
    fun getFeeds(): Call<PostDataModel>

}